tests/utils/campaignService.test.js

```diff
@@ -0,0 +1,19 @@
+import { getFreeDomain } from "../../utils/campaignService";
+
+global.fetch = jest.fn().mockResolvedValue({
+  status: 200,
+  json: () => ({
+    r: "0x1234567890123456789012345678901234567890",
+    s: "0x1234567890123456789012345678901234567890",
+  }),
+});
+
+describe("getFreeDomain function", () => {
```

Comment:
what if it doesn't return the expected result. You should test all cases including eroors ...